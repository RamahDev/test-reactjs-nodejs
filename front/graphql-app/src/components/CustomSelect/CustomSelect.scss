.form-control{
    .MuiButton-root, .MuiFormLabel-root, .MuiInputBase-input, .MuiIconButton-root, .MuiInputBase-root, .MuiSelect-icon{
        color:#fff;
    }
    .MuiInput-underline:before, .MuiInput-underline:hover:not(.Mui-disabled):before{
        border-bottom: 1px solid #fff;
    }
    .MuiFormLabel-root.Mui-focused {
        color:#fff;
    }
    .MuiInput-underline:after {
        border-bottom: 2px solid #fff;
    }
}