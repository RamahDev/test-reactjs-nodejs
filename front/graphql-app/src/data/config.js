

let url = 'http://localhost:4000/';

export default {
  url,
  baseURL: 'http://localhost:4000/',
  apiDocs: `${url}api-docs/`,
  pathApi: 'api/',
  myprivatekey: 'myprivatekey',
  FACEBOOK_APP_ID: '873298023421774',
  GOOGLE_APP_ID:
    '405680342370-r0ktkaio1g124q4kq39qqa35p7u6lnnr.apps.googleusercontent.com',
};
