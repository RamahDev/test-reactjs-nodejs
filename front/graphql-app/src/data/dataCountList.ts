export const dataCountList = [
  {
    title: "30",
    value: "30",
  },
  {
    title: "50",
    value: "50",
  },
  {
    title: "100",
    value: "100",
  },
  {
    title: "All",
    value: "all",
  },
];

export const dataCron = [
  {
    title: "En cours",
    value: 1,
  },
  {
    title: "Terminer",
    value: 0,
  },
];

export const dataStream = [
  {
    title: "Top",
    value: "top",
  },
  {
    title: "Debutants",
    value: "debutants",
  },
];

export const timeViewers = [
  {
    title: "10s",
    value: "10s",
  },
  {
    title: "15s",
    value: "15s",
  },
  {
    title: "30s",
    value: "30s",
  },
  {
    title: "45s",
    value: "45s",
  },
  {
    title: "1 min",
    value: "1min",
  },
  {
    title: "2 min",
    value: "2min",
  },
  {
    title: "5 min",
    value: "5min",
  },
  {
    title: "10 min",
    value: "10min",
  },
  {
    title: "30 min",
    value: "30min",
  },
  {
    title: "45 min",
    value: "45min",
  },
  {
    title: "1 h",
    value: "1h",
  },
  {
    title: "2h",
    value: "2h",
  },
  {
    title: "3h",
    value: "3h",
  },
];

export const paginationOptions = [
  {
    title: "10",
    value: 10,
  },
  {
    title: "20",
    value: 20,
  },
  {
    title: "50",
    value: 50,
  },
  {
    title: "100",
    value: 100,
  },
  {
    title: "All",
    value: 1000000,
  },
];
