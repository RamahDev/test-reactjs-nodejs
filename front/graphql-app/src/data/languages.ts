export const languages = [
  {
    title: "Tout",
    value: "all",
  },
  {
    title: "Anglais",
    value: "en",
  },
  {
    title: "Français",
    value: "fr",
  },
  {
    title: "Epagnol",
    value: "es",
  },
  {
    title: "Allemand",
    value: "de",
  },
  {
    title: "Italien",
    value: "it",
  },
  {
    title: "Portugais",
    value: "pt",
  },
  {
    title: "Japonais",
    value: "ja",
  },
  {
    title: "Chinois",
    value: "zh",
  },
  {
    title: "Hindi",
    value: "hi",
  },
  {
    title: "Arabe",
    value: "ar",
  },
  {
    title: "Coréen",
    value: "ko",
  },
  {
    title: "Tchèque",
    value: "cs",
  },
  {
    title: "Danoise",
    value: "da",
  },
  {
    title: "Slovaque",
    value: "sk",
  },
  {
    title: "Turc",
    value: "tr",
  },
  {
    title: "Néerlandaise",
    value: "nl",
  },
  {
    title: "Grecque",
    value: "el",
  },
  {
    title: "Hongroise",
    value: "hu",
  },
  {
    title: "Indonésienne",
    value: "id",
  },
  {
    title: "Norvégienne",
    value: "no",
  },
  {
    title: "Roumaine",
    value: "ro",
  },
  {
    title: "Russe",
    value: "ru",
  },
  {
    title: "Suédoise",
    value: "sv",
  },
  {
    title: "Thaïlandais",
    value: "th",
  },
  {
    title: "Ukrainienne",
    value: "uk",
  },
  {
    title: "Vietnamienne",
    value: "vi",
  },
];
