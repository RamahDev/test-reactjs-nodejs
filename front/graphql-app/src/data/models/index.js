import {capitalizeFirstLetter} from '../../services/technique/utils';

const data = require('./Fragment.json');

const toDefault = (type, defaultValue) => {
  switch (type) {
    case 'String':
    case 'string':
    case String:
      return {
        type: String,
        default: defaultValue || '',
      };
    case Array:
    case 'Array':
      return {
        type: Array,
        default: defaultValue || [],
      };
    case Object:
    case 'Object':
      return {
        type: Object,
        default: defaultValue || {},
      };
    default:
      return {
        type: String,
        default: defaultValue || '',
      };
  }
};
const initialAttributs = name => ({
  ['list' + capitalizeFirstLetter(name)]: {
    type: Array,
    default: [],
  },
  [name]: {
    type: Object,
    default: {},
  },
});

export const UMLClass =
  data && data.ownedElements
    ? data.ownedElements
        .filter(i => i._type === 'UMLClass')
        .map(className => {
          const {attributes, operations} = className;
          const name = className.name.toLowerCase();
          const model = operations
            ? operations.map(operation => {
                const {specification, ...all} = operation;
                let urlAndMethod = {};
                const documentation = specification;
                if (
                  documentation &&
                  documentation.split('&&') &&
                  documentation.split('&&').length > 0
                ) {
                  urlAndMethod = documentation
                    .split('&&')
                    .reduce((acc, curr) => {
                      const val = curr.split('=');
                      if (val && val.length > 1) {
                        return {
                          ...acc,
                          [val[0]]: val[1],
                        };
                      }
                      return acc;
                    }, {});
                }
                return {
                  ...all,
                  ...urlAndMethod,
                };
              })
            : [];

          const attributs = attributes.reduce((acc, curr) => {
            const {name, type, documentation} = curr;
            let defaultValue;
            if (documentation) {
              defaultValue = documentation;
            }
            return {
              ...acc,
              [name]: toDefault(type, defaultValue),
            };
          }, initialAttributs(name));
          return {
            ...className,
            model,
            entity: name,
            name,
            attributs,
          };
        })
    : [];

const objectClass = UMLClass.reduce((acc, curr) => {
  const {name} = curr;
  return {
    ...acc,
    [name]: curr,
  };
}, {});

export default objectClass;
