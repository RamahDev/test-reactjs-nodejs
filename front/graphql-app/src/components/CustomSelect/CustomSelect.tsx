import React from "react";
import { makeStyles, TextField } from "@material-ui/core";
import { Autocomplete } from "@material-ui/lab";
import { dataCountList, dataStream } from "../../data/dataCountList";
import { CustomSelectProps } from "../../models/CustomSelectProps";

import "./CustomSelect.scss";

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  autoComplete: {
    width: "150px",
    color: "white",
    textAlign: "center",
  },
  formControl: {
    margin: theme.spacing(1),
    minWidth: 120,
  },
}));

const CustomSelect: React.FC<CustomSelectProps> = ({
  dataCount,
  setDataCount,
  isStreamer,
  stream,
  setStream,
}) => {
  const classes = useStyles();

  return (
    <>
      <Autocomplete
        className={`${classes.autoComplete} auto-complete`}
        id="size-small-standard"
        size="small"
        onChange={(event, newValue) => {
          if (newValue) {
            if (isStreamer) {
              setStream(newValue);
            } else {
              setDataCount(newValue);
            }
          }
        }}
        options={isStreamer ? dataStream : dataCountList}
        getOptionLabel={(option: any) => option.title || ""}
        getOptionSelected={(option: any) => option.title}
        value={isStreamer ? stream : dataCount}
        renderInput={(params: any) => (
          <TextField
            {...params}
            variant="standard"
            placeholder="DonnÃ©es par scroll"
          />
        )}
      />
    </>
  );
};

export default CustomSelect;
