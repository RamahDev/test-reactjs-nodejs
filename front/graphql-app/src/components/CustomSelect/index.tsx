import { makeStyles, TextField } from "@material-ui/core";
import { Autocomplete } from "@material-ui/lab";
import React from "react";
import { SelectProps } from "../../models/CustomSelectProps";
import "./CustomSelect.scss";

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  autoComplete: {
    width: "150px",
    color: "white",
    textAlign: "center",
  },
  formControl: {
    margin: theme.spacing(1),
    minWidth: 120,
  },
}));

const Select: React.FC<SelectProps> = ({
  item,
  setItem,
  options
}) => {
  const classes = useStyles();

  return (
    <>
      <Autocomplete
        className={`${classes.autoComplete} auto-complete`}
        id="size-small-standard"
        size="small"
        onChange={(event, newValue) => {
          if (newValue) {
            setItem(newValue);
          }
        }}
        options={options}
        getOptionLabel={(option: any) => option.title}
        getOptionSelected={(option: any) => option.title}
        value={item}
        renderInput={(params: any) => (
          <TextField
            {...params}
            variant="standard"
            placeholder="DonnÃ©es par scroll"
          />
        )}
      />
    </>
  );
};

export default Select;
