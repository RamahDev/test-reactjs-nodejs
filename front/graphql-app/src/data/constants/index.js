import models from '../models';
import {
  toGenerateInitialSate,
  toGenerateConstantsArray,
} from '../../services/technique/generateCode';

export default Object.values(models).map(item => {
  const {attributs, model, entity} = item;
  return {
    entity,
    constants: toGenerateConstantsArray(model, entity),
    initialState: toGenerateInitialSate(attributs),
  };
});

export const constants = Object.values(models).map(item => {
  const {attributs, model, entity} = item;
  const res = {
    entity,
    constants: toGenerateConstantsArray(model, entity),
    initialState: toGenerateInitialSate(attributs),
  };
  return {[entity]: res};
});
