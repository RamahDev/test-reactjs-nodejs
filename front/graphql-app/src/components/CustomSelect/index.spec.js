import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import CustomSelect from './CustomSelect';

import { dataCountList, dataStream } from "../../data/dataCountList";

const toggleExpand = jest.fn();
const defaultProps = {
  setDataCount: toggleExpand,
  dataCount: [],
  isStreamer: true,
  stream: [],
  setStream: [],
  dataCountList,
  dataStream,
  onClick: toggleExpand,
};

test('<CustomSelect /> component renders correctly', () => {
  const { container } = render(<CustomSelect {...defaultProps} />);
  expect(container).toMatchSnapshot();
});
