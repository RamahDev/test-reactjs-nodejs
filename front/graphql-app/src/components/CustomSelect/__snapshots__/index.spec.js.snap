// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomSelect /> component renders correctly 1`] = `
<div>
  <div
    aria-expanded="false"
    class="MuiAutocomplete-root makeStyles-autoComplete-2 auto-complete MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
    role="combobox"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd MuiInputBase-marginDense MuiInput-marginDense"
      >
        <input
          aria-autocomplete="list"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
          id="size-small-standard"
          placeholder="DonnÃ©es par scroll"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Clear"
            class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
            tabindex="-1"
            title="Clear"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
